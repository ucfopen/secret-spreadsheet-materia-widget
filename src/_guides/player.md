## Overview

Students must fill in the blanks on a spreadsheet. The blanks can be either randomized or fixed.

![player screen](assets/player.png)

## Details

### Filling in the blanks

Blanks that still need to be answered will be highlighted in blue and will have a question mark. Blanks that have been filled in will be highlighted in yellow. There is a running tally of how many blanks have been filled in out of all of the blanks at the bottom.

### Question and description

There may be an accompanying question, description, or instructions that pop up before you are able to use the spreadsheet. If you want to read it again, click the **View Question** text.

### Submitting

When you have filled in all then blanks that you want to, click on the **Submit** button. You will be scored based on how many blanks were filled with the correct information, spelling, capitalization, and spaces do matter. 