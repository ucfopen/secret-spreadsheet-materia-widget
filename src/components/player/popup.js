import React from 'react';

// popup containing instructions on how to use the wdget
const Popup = props => {
	return(
		<div className="popup help">
			<div>
				<h2>Secret Spreadsheet - How it works:</h2>

				<div>
					<img src="assets/img/helper-pic.svg" alt="Secret Spreadsheet Instructions" />

					<p>Cells of this spreadsheet have been <strong>intentionally hidden.</strong></p>

					<p>Input the hidden data correctly <strong>(capitalization matters!)</strong> to restore the spreadsheet.</p>
				</div>

				<button type="button" value="Got it!" onClick={props.handlePopupToggle}>
					Got it!
				</button>
			</div>
		</div>
	);
};

export default Popup;
