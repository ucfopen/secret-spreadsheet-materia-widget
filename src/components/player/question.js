import React from 'react';

// the popup containing the question and description given by the widget creator
// at a minimum shows the question, description is optional
const Question = props => {
	return (
		<div className="popup question">
			<div>
				<p className="mainQuestion">{props.question}</p>

				{props.description !== '' ? <p>{props.description}</p>:null}

				<button type="button" value="Next" onClick={props.handleQuestionToggle}>
						Next
				</button>
			</div>
		</div>
	);
};

export default Question;
