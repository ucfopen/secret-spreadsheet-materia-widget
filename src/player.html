<!DOCTYPE html>
<html>
	<head>
		<title>Spreadsheet Widget Player</title>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />

		<link rel="stylesheet" type="text/css" href="player.css">

		<script src="materia.enginecore.js"></script>

		<script src="//cdnjs.cloudflare.com/ajax/libs/react/16.5.2/umd/react.development.js"></script>
		<script src="//cdnjs.cloudflare.com/ajax/libs/react-dom/16.5.2/umd/react-dom.development.js"></script>
	</head>
	<body>
		<div id="root"></div>
		<script src="player.js"></script>
	</body>
</html>
