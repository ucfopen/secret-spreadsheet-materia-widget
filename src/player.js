import React from 'react';
import ReactDOM from 'react-dom';
import PlayerTable from './components/player-table';
import Popup from './components/player-popup';

class PlayerApp extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			popup: true,
			answered: 0
		}
		this.answers = {};
		this.blankPositions = new Set(); // list of blank cells
		this.submitAnswer = this.submitAnswer.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleNewAnswer = this.handleNewAnswer.bind(this);
		this.handlePopupToggle = this.handlePopupToggle.bind(this);
		this.randomize = this.randomize.bind(this);
		this.countBlanks = this.countBlanks.bind(this);
	}

	// check if question has been answered. If it is, submit the answer, or submit blank
	submitAnswer(id, counter) {
		if (this.answers.hasOwnProperty(`${counter}-input`)) {
			Materia.Score.submitQuestionForScoring(id, this.answers[`${counter}-input`]);
		}
		else {
			Materia.Score.submitQuestionForScoring(id, '');
		}
	}

	// check either the answers that were marked as blank or randomly selected as blank
	handleSubmit(event) {
		let counter = 0;

		event.preventDefault();

		this.props.qset.forEach(element => {
			element.forEach(question => {
				// randomly selected questions
				if (this.props.randCount !== 0) {
					if (this.blankPositions.has(counter)) {
						this.submitAnswer(question.id, counter);
					}
				}
				// not random
				else {
					if (question.options.blank) {
						this.submitAnswer(question.id, counter);
					}
				}

				counter++;
			});
		});

		Materia.Engine.end();
	}

	// add new answers to bank and if the user filled in a blank for the first time increment the answered question count
	handleNewAnswer(newAnswers, fromBlankToFilled, fromFilledToBlank) {
		this.answers = newAnswers;

		if (fromBlankToFilled && !fromFilledToBlank && this.state.answered < this.blankPositions.size) {
			this.setState({
				popup: this.state.popup,
				answered: this.state.answered + 1
			});
		}
		else if (!fromBlankToFilled && fromFilledToBlank && this.state.answered > 0) {
			this.setState({
				popup: this.state.popup,
				answered: this.state.answered - 1
			});
		}
		else if (fromBlankToFilled && fromFilledToBlank) {
			console.error('Answer count error in handleNewAnswer');
		}
	}

	// decides if it should show popup or not
	handlePopupToggle() {
		if (this.state.popup) {
			this.setState({
				popup: false
			});
		}
		else {
			this.setState({
				popup: true
			});
		}
	}

	// select random blank answers
	randomize() {
		let totalCells = this.props.dimensions.x * this.props.dimensions.y;
		let selectCount = 0;

		if (this.props.header) {
			totalCells -= this.props.dimensions.x;
		}

		while (selectCount < this.props.randCount) {
			const position = Math.floor(Math.random() * totalCells + (this.props.header ? this.props.dimensions.x:0));

			if (!this.blankPositions.has(position)) {
				this.blankPositions.add(position);
				selectCount++;
			}
		}
	}

	// as the table is being created from user selected blanks, add each blank to the blank position set
	countBlanks(position) {
		this.blankPositions.add(position);
	}

	render() {
		// randomize which entries are blank if the creator says more than 0 should be random and this is the first render of the table
		if (this.blankPositions.size === 0 && this.props.randCount !== 0) {
			this.randomize();
		}

		return(
			<div>
				<header>
					<h1>{this.props.title}</h1>
					{this.state.popup ? null:<button type="button" value="Help" onClick={this.handlePopupToggle}>Help</button>}
				</header>

				<main>
					{this.state.popup ? <Popup handlePopupToggle={this.handlePopupToggle} />:null}

					<p className="instructions">Input the <span><strong>missing data</strong></span> to complete the spreadsheet:</p>

					<form onSubmit={this.handleSubmit}>
						<table>
							<tbody>
								<PlayerTable
									dimensions={this.props.dimensions}
									qset={this.props.qset}
									parentAnswers={this.answers}
									handleNewAnswer={this.handleNewAnswer}
									randPositions={this.blankPositions}
									randCount={this.props.randCount}
									countBlanks={this.countBlanks}
									leftAlign={this.props.leftAlign}
									header={this.props.header}
								/>
							</tbody>
						</table>

						<p>You've filled out {this.state.answered} of {this.blankPositions.size} missing cells</p>

						<input
							type="submit"
							value="Submit"
							className={this.state.answered !== this.blankPositions.size ? 'grayed':'filled'} />
					</form>
				</main>
			</div>
		);
	}
}

Materia.Engine.start({
	start: (instance, qset) => {
		ReactDOM.render(
			<PlayerApp
				title={instance.name}
				dimensions={qset.dimensions}
				qset={qset.items[0].items}
				randCount={qset.randomization}
				leftAlign={qset.left}
				header={qset.header}
			/>,
			document.getElementById('root')
		);
	},
	manualResize: false
});
