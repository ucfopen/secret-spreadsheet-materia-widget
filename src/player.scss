@import 'base';

h1 {
	margin: 20px auto;
}

input {
	&:hover, &:focus {
		background-color: inherit;
	}
}

table {
	max-width: 712px;
	border-collapse: collapse;
	margin: 40px auto;
}
