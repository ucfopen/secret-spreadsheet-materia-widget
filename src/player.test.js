import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from './enzyme'
import ReactDOM from 'react-dom';

const mockDomRender = jest.fn();

const qset = () => {
	return {
		"left": false,
		"header": false,
		"spreadsheet": false,
		"dimensions": {
			"x": 2,
			"y": 2
		},
		"randomization": 0,
		"entryQuestion": "Test Question",
		"questionBody": "Test question body",
		"items": [
			{
				"items": [
					[
						{
							"materiaType": "question",
							"id": null,
							"type": "QA",
							"questions": [
								{
									"text": "Show 1"
								}
							],
							"answers": [
								{
									"id": null,
									"text": "Show 1",
									"value": 100
								}
							],
							"options": {
								"blank": false
							}
						},
						{
							"materiaType": "question",
							"id": null,
							"type": "QA",
							"questions": [
								{
									"text": "Hide 1"
								}
							],
							"answers": [
								{
									"id": null,
									"text": "Hide 1",
									"value": 100
								}
							],
							"options": {
								"blank": true
							}
						}
					],
					[
						{
							"materiaType": "question",
							"id": null,
							"type": "QA",
							"questions": [
								{
									"text": "Hide 2"
								}
							],
							"answers": [
								{
									"id": null,
									"text": "Hide 2",
									"value": 100
								}
							],
							"options": {
								"blank": true
							}
						},
						{
							"materiaType": "question",
							"id": null,
							"type": "QA",
							"questions": [
								{
									"text": "Show 2"
								}
							],
							"answers": [
								{
									"id": null,
									"text": "Show 2",
									"value": 100
								}
							],
							"options": {
								"blank": false
							}
						}
					]
				]
			}
		]
	};
};

const genProps = () => {
	const tempQset = qset();
	return {
		title: 'Test Widget',
		dimensions: tempQset.dimensions,
		qset: tempQset.items[0].items,
		randCount: tempQset.randomization,
		leftAlign: tempQset.left,
		header: tempQset.header,
		spreadsheet: tempQset.spreadsheet,
		entryQuestion: tempQset.entryQuestion,
		questionBody: tempQset.questionBody
	};
};

const makeNewPlayer = (changes={}) => {
	const PlayerApp = require('./player').default;

	const props = genProps();
	Object.defineProperties(props, changes);

	return shallow(<PlayerApp {... props} />);
};

describe('Player', () => {
	beforeEach(() => {
		jest.resetAllMocks();
		jest.resetModules();
		jest.unmock('react-dom');
		global.Materia = {
			Engine: {
				start: jest.fn(),
				end: jest.fn()
			},
			Score: {
				submitQuestionForScoring: jest.fn()
			}
		};
	});

	test('Registers callbacks', () => {
		require('./player');
		jest.mock('react-dom', () => ({
			render: jest.fn()
		}));

		expect(Materia.Engine.start).toHaveBeenCalledTimes(1);
		const callbacks = Materia.Engine.start.mock.calls[0][0];
		expect(callbacks).toHaveProperty('start', expect.any(Function));
	});

	test('Render called with correct props no randomization', () => {
		require('./player');
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}));

		const start = Materia.Engine.start.mock.calls[0][0].start;

		expect(mockDomRender).toHaveBeenCalledTimes(0);
		start({name: 'Test Widget'}, qset());
		expect(mockDomRender).toHaveBeenCalledTimes(1);
		expect(mockDomRender.mock.calls[0][0]).toMatchSnapshot();
	});

	test('Render called with correct props with randomization', () => {
		require('./player');
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}));

		const start = Materia.Engine.start.mock.calls[0][0].start;
		const newQset = qset();
		newQset.randomization = 2;

		expect(mockDomRender).toHaveBeenCalledTimes(0);
		start({name: 'Test Widget'}, newQset);
		expect(mockDomRender).toHaveBeenCalledTimes(1);
		expect(mockDomRender.mock.calls[0][0]).toMatchSnapshot();
	});

	// can't test rendering with randomization because it is changed every time
	// randomization function test will ensure the randomization is working
	test('Render without randomization', () => {
		require('./player');
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}));

		const start = Materia.Engine.start.mock.calls[0][0].start;
		start({name: 'Test Widget'}, qset());
		const playerComponent = mockDomRender.mock.calls[0][0];

		const component = renderer.create(playerComponent);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Render with question and body', () => {
		require('./player');
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}));

		const start = Materia.Engine.start.mock.calls[0][0].start;
		start({name: 'Test Widget'}, qset());
		const playerComponent = mockDomRender.mock.calls[0][0];

		const component = renderer.create(playerComponent);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Render with question and no body', () => {
		require('./player');
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}));

		const start = Materia.Engine.start.mock.calls[0][0].start;
		const newQset = qset();
		newQset.questionBody = '';
		start({name: 'Test Widget'}, newQset);
		const playerComponent = mockDomRender.mock.calls[0][0];

		const component = renderer.create(playerComponent);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Render with no question', () => {
		require('./player');
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}));

		const start = Materia.Engine.start.mock.calls[0][0].start;
		const newQset = qset();
		newQset.entryQuestion = '';
		newQset.questionBody = '';
		start({name: 'Test Widget'}, newQset);
		const playerComponent = mockDomRender.mock.calls[0][0];

		const component = renderer.create(playerComponent);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('submitAnswer given user entered answer', () => {
		const mockPlayer = makeNewPlayer();
		mockPlayer.instance().answers['0-input'] = 'TestAnswer';

		mockPlayer.instance().submitAnswer(1, 0);

		expect(Materia.Score.submitQuestionForScoring).toHaveBeenCalledTimes(1);
		expect(Materia.Score.submitQuestionForScoring).toHaveBeenCalledWith(1, 'TestAnswer');

		// cleanup
		mockPlayer.unmount();
	});

	test('submitAnswer given no user input', () => {
		const mockPlayer = makeNewPlayer();

		mockPlayer.instance().submitAnswer(1, 0);

		expect(Materia.Score.submitQuestionForScoring).toHaveBeenCalledTimes(1);
		expect(Materia.Score.submitQuestionForScoring).toHaveBeenCalledWith(1, '');

		// cleanup
		mockPlayer.unmount();
	});

	test('handleSubmit works without randomization', () => {
		const event = {
			preventDefault: jest.fn()
		}
		const mockPlayer = makeNewPlayer();
		mockPlayer.instance().submitAnswer = jest.fn();

		mockPlayer.instance().handleSubmit(event);

		expect(event.preventDefault).toHaveBeenCalled();
		expect(mockPlayer.instance().submitAnswer).toHaveBeenCalledTimes(2);
		expect(mockPlayer.instance().submitAnswer).toHaveBeenNthCalledWith(1, null, 1);
		expect(mockPlayer.instance().submitAnswer).toHaveBeenNthCalledWith(2, null, 2);
		expect(Materia.Engine.end).toHaveBeenCalled();

		// cleanup
		mockPlayer.unmount();
	});

	test('handleSubmit works with randomization', () => {
		const event = {
			preventDefault: jest.fn()
		}
		const mockPlayer = makeNewPlayer({randCount: {value: 2}});
		mockPlayer.instance().submitAnswer = jest.fn();
		mockPlayer.instance().blankPositions = new Set([1, 2]);

		mockPlayer.instance().handleSubmit(event);

		expect(event.preventDefault).toHaveBeenCalled();
		expect(mockPlayer.instance().submitAnswer).toHaveBeenCalledTimes(2);
		expect(mockPlayer.instance().submitAnswer).toHaveBeenNthCalledWith(1, null, 1);
		expect(mockPlayer.instance().submitAnswer).toHaveBeenNthCalledWith(2, null, 2);
		expect(Materia.Engine.end).toHaveBeenCalled();
	});

	test('handleNewAnswer adding new answer from blank to filled', () => {
		const mockPlayer = makeNewPlayer();
		const newAnswers = {'TestID': 'TestAnswer'};

		mockPlayer.setState({
			popup: false,
			answered: 0
		});
		mockPlayer.instance().blankPositions.add(1);

		mockPlayer.instance().handleNewAnswer(newAnswers, true, false);

		expect(mockPlayer.instance().answers.hasOwnProperty('TestID')).toBeTruthy();
		expect(mockPlayer.instance().answers.TestID).toBe('TestAnswer');
		expect(mockPlayer.state(['popup'])).toBeFalsy();
		expect(mockPlayer.state(['answered'])).toBe(1);

		// cleanup
		mockPlayer.unmount();
	});

	test('handleNewAnswer removing an answer from filled to blank', () => {
		const mockPlayer = makeNewPlayer();
		const newAnswers = {'TestID': ''};

		mockPlayer.setState({
			popup: false,
			answered: 1
		});

		mockPlayer.instance().handleNewAnswer(newAnswers, false, true);

		expect(mockPlayer.instance().answers.hasOwnProperty('TestID')).toBeTruthy();
		expect(mockPlayer.instance().answers.TestID).toBe('');
		expect(mockPlayer.state(['popup'])).toBeFalsy();
		expect(mockPlayer.state(['answered'])).toBe(0);

		// cleanup
		mockPlayer.unmount();
	});

	test('handleNewAnswer changing an answer', () => {
		const mockPlayer = makeNewPlayer();
		const newAnswers = {'TestID': 'Changed'};

		mockPlayer.setState({
			popup: false,
			answered: 1
		});
		mockPlayer.instance().answers = {'TestID': 'Original'};

		mockPlayer.instance().handleNewAnswer(newAnswers, false, false);

		expect(mockPlayer.instance().answers.hasOwnProperty('TestID')).toBeTruthy();
		expect(mockPlayer.instance().answers.TestID).toBe('Changed');
		expect(mockPlayer.state(['popup'])).toBeFalsy();
		expect(mockPlayer.state(['answered'])).toBe(1);
	});

	test('handleNewAnswer is given both empty to full and full to empty answers', () => {
		const mockPlayer = makeNewPlayer();
		const newAnswers = {};

		// capture errors so that it doesn't go to the console
		const realError = console.error;
		console.error = jest.fn();

		mockPlayer.instance().handleNewAnswer(newAnswers, true, true);

		expect(console.error).toHaveBeenCalled();

		// cleanup
		console.error = realError;
		mockPlayer.unmount();
	});

	test('handlePopupToggle is correctly toggled off', () => {
		const mockPlayer = makeNewPlayer();

		mockPlayer.setState({
			popup: true,
			answered: 0
		});
		mockPlayer.instance().handlePopupToggle();

		expect(mockPlayer.state(['popup'])).toBeFalsy();

		// cleanup
		mockPlayer.unmount();
	});

	test('handlePopupToggle is toggled on', () => {
		const mockPlayer = makeNewPlayer();

		mockPlayer.setState({
			popup: false,
			answered: 0
		});
		mockPlayer.instance().handlePopupToggle();

		expect(mockPlayer.state(['popup'])).toBeTruthy();

		// cleanup
		mockPlayer.unmount();
	});

	test('handleQuestionToggle is toggled off', () => {
		const mockPlayer = makeNewPlayer();

		mockPlayer.setState({
			question: true,
			showQuestion: true
		});
		mockPlayer.instance().handleQuestionToggle();

		expect(mockPlayer.state(['showQuestion'])).toBeFalsy();

		// cleanup
		mockPlayer.unmount();
	});

	test('handleQuestionToggle is toggled on', () => {
		const mockPlayer = makeNewPlayer();

		mockPlayer.setState({
			question: true,
			showQuestion: false
		});
		mockPlayer.instance().handleQuestionToggle();

		expect(mockPlayer.state(['showQuestion'])).toBeTruthy();

		// cleanup
		mockPlayer.unmount();
	});

	test('randomize works without header', () => {
		const mockPlayer = makeNewPlayer({randCount: {value: 2}});
		mockPlayer.instance().blankPositions = new Set();

		mockPlayer.instance().randomize();

		expect(mockPlayer.instance().blankPositions.size).toBe(2);

		// cleanup
		mockPlayer.unmount();
	});

	test('randomize works with header', () => {
		const mockPlayer = makeNewPlayer({
			randCount: {value: 1},
			header: {value: true}
		});
		mockPlayer.instance().blankPositions = new Set();

		mockPlayer.instance().randomize();

		expect(mockPlayer.instance().blankPositions.size).toBe(1);
		expect(mockPlayer.instance().blankPositions.has(0)).toBeFalsy();
		expect(mockPlayer.instance().blankPositions.has(1)).toBeFalsy();

		// cleanup
		mockPlayer.unmount();
	});

	test('countBlanks called and added', () => {
		const mockPlayer = makeNewPlayer();

		mockPlayer.instance().countBlanks(1);

		expect(mockPlayer.instance().blankPositions.has(1)).toBeTruthy();

		// cleanup
		mockPlayer.unmount();
	});
});
