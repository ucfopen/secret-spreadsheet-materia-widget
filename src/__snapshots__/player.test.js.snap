// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Render called with correct props no randomization 1`] = `
<PlayerApp
  description="Test question body"
  dimensions={
    Object {
      "x": 2,
      "y": 2,
    }
  }
  header={false}
  leftAlign={false}
  qset={
    Array [
      Array [
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Show 1",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": false,
          },
          "questions": Array [
            Object {
              "text": "Show 1",
            },
          ],
          "type": "QA",
        },
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Hide 1",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": true,
          },
          "questions": Array [
            Object {
              "text": "Hide 1",
            },
          ],
          "type": "QA",
        },
      ],
      Array [
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Hide 2",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": true,
          },
          "questions": Array [
            Object {
              "text": "Hide 2",
            },
          ],
          "type": "QA",
        },
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Show 2",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": false,
          },
          "questions": Array [
            Object {
              "text": "Show 2",
            },
          ],
          "type": "QA",
        },
      ],
    ]
  }
  question="Test Question"
  randCount={0}
  spreadsheet={false}
  title="Test Widget"
/>
`;

exports[`Player Render called with correct props with randomization 1`] = `
<PlayerApp
  description="Test question body"
  dimensions={
    Object {
      "x": 2,
      "y": 2,
    }
  }
  header={false}
  leftAlign={false}
  qset={
    Array [
      Array [
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Show 1",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": false,
          },
          "questions": Array [
            Object {
              "text": "Show 1",
            },
          ],
          "type": "QA",
        },
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Hide 1",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": true,
          },
          "questions": Array [
            Object {
              "text": "Hide 1",
            },
          ],
          "type": "QA",
        },
      ],
      Array [
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Hide 2",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": true,
          },
          "questions": Array [
            Object {
              "text": "Hide 2",
            },
          ],
          "type": "QA",
        },
        Object {
          "answers": Array [
            Object {
              "id": null,
              "text": "Show 2",
              "value": 100,
            },
          ],
          "id": null,
          "materiaType": "question",
          "options": Object {
            "blank": false,
          },
          "questions": Array [
            Object {
              "text": "Show 2",
            },
          ],
          "type": "QA",
        },
      ],
    ]
  }
  question="Test Question"
  randCount={2}
  spreadsheet={false}
  title="Test Widget"
/>
`;

exports[`Player Render with no question 1`] = `
<div>
  <header>
    <h1>
      Test Widget
    </h1>
  </header>
  <main>
    <div
      className="popup help"
    >
      <div>
        <h2>
          Secret Spreadsheet - How it works:
        </h2>
        <div>
          <img
            alt="Secret Spreadsheet Instructions"
            src="assets/img/helper-pic.svg"
          />
          <p>
            Cells of this spreadsheet have been 
            <strong>
              intentionally hidden.
            </strong>
          </p>
          <p>
            Input the hidden data correctly 
            <strong>
              (capitalization matters!)
            </strong>
             to restore the spreadsheet.
          </p>
        </div>
        <button
          onClick={[Function]}
          type="button"
          value="Got it!"
        >
          Got it!
        </button>
      </div>
    </div>
    <p
      className="instructions"
    >
      Input the 
      <span>
        missing data
      </span>
       to complete the spreadsheet:
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="table-surround"
      >
        <div>
          <table>
            <thead />
            <tbody>
              <tr
                id="row0"
              >
                <td
                  className="centerAlign"
                  id="cell0"
                >
                  <span>
                    Show 1
                  </span>
                </td>
                <td
                  className="unanswered centerAlign"
                  id="cell1"
                >
                  <input
                    className="unanswered"
                    id="1-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
              </tr>
              <tr
                id="row1"
              >
                <td
                  className="unanswered centerAlign"
                  id="cell2"
                >
                  <input
                    className="unanswered"
                    id="2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
                <td
                  className="centerAlign"
                  id="cell3"
                >
                  <span>
                    Show 2
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <p>
        You've filled out 
        0
         of 
        0
         missing cells
      </p>
      <input
        className="filled"
        type="submit"
        value="Submit"
      />
    </form>
  </main>
</div>
`;

exports[`Player Render with question and body 1`] = `
<div>
  <header>
    <h1>
      Test Widget
    </h1>
  </header>
  <main>
    <div
      className="popup help"
    >
      <div>
        <h2>
          Secret Spreadsheet - How it works:
        </h2>
        <div>
          <img
            alt="Secret Spreadsheet Instructions"
            src="assets/img/helper-pic.svg"
          />
          <p>
            Cells of this spreadsheet have been 
            <strong>
              intentionally hidden.
            </strong>
          </p>
          <p>
            Input the hidden data correctly 
            <strong>
              (capitalization matters!)
            </strong>
             to restore the spreadsheet.
          </p>
        </div>
        <button
          onClick={[Function]}
          type="button"
          value="Got it!"
        >
          Got it!
        </button>
      </div>
    </div>
    <div
      className="popup question"
    >
      <div>
        <p
          className="mainQuestion"
        >
          Test Question
        </p>
        <p>
          Test question body
        </p>
        <button
          onClick={[Function]}
          type="button"
          value="Next"
        >
          Next
        </button>
      </div>
    </div>
    <p
      className="instructions"
    >
      Input the 
      <span>
        missing data
      </span>
       to complete the spreadsheet:
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="table-surround"
      >
        <div>
          <table>
            <thead />
            <tbody>
              <tr
                id="row0"
              >
                <td
                  className="centerAlign"
                  id="cell0"
                >
                  <span>
                    Show 1
                  </span>
                </td>
                <td
                  className="unanswered centerAlign"
                  id="cell1"
                >
                  <input
                    className="unanswered"
                    id="1-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
              </tr>
              <tr
                id="row1"
              >
                <td
                  className="unanswered centerAlign"
                  id="cell2"
                >
                  <input
                    className="unanswered"
                    id="2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
                <td
                  className="centerAlign"
                  id="cell3"
                >
                  <span>
                    Show 2
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <p>
        You've filled out 
        0
         of 
        0
         missing cells
      </p>
      <p
        className="link"
        onClick={[Function]}
      >
        View Question
      </p>
      <input
        className="filled"
        type="submit"
        value="Submit"
      />
    </form>
  </main>
</div>
`;

exports[`Player Render with question and no body 1`] = `
<div>
  <header>
    <h1>
      Test Widget
    </h1>
  </header>
  <main>
    <div
      className="popup help"
    >
      <div>
        <h2>
          Secret Spreadsheet - How it works:
        </h2>
        <div>
          <img
            alt="Secret Spreadsheet Instructions"
            src="assets/img/helper-pic.svg"
          />
          <p>
            Cells of this spreadsheet have been 
            <strong>
              intentionally hidden.
            </strong>
          </p>
          <p>
            Input the hidden data correctly 
            <strong>
              (capitalization matters!)
            </strong>
             to restore the spreadsheet.
          </p>
        </div>
        <button
          onClick={[Function]}
          type="button"
          value="Got it!"
        >
          Got it!
        </button>
      </div>
    </div>
    <div
      className="popup question"
    >
      <div>
        <p
          className="mainQuestion"
        >
          Test Question
        </p>
        <button
          onClick={[Function]}
          type="button"
          value="Next"
        >
          Next
        </button>
      </div>
    </div>
    <p
      className="instructions"
    >
      Input the 
      <span>
        missing data
      </span>
       to complete the spreadsheet:
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="table-surround"
      >
        <div>
          <table>
            <thead />
            <tbody>
              <tr
                id="row0"
              >
                <td
                  className="centerAlign"
                  id="cell0"
                >
                  <span>
                    Show 1
                  </span>
                </td>
                <td
                  className="unanswered centerAlign"
                  id="cell1"
                >
                  <input
                    className="unanswered"
                    id="1-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
              </tr>
              <tr
                id="row1"
              >
                <td
                  className="unanswered centerAlign"
                  id="cell2"
                >
                  <input
                    className="unanswered"
                    id="2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
                <td
                  className="centerAlign"
                  id="cell3"
                >
                  <span>
                    Show 2
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <p>
        You've filled out 
        0
         of 
        0
         missing cells
      </p>
      <p
        className="link"
        onClick={[Function]}
      >
        View Question
      </p>
      <input
        className="filled"
        type="submit"
        value="Submit"
      />
    </form>
  </main>
</div>
`;

exports[`Player Render without randomization 1`] = `
<div>
  <header>
    <h1>
      Test Widget
    </h1>
  </header>
  <main>
    <div
      className="popup help"
    >
      <div>
        <h2>
          Secret Spreadsheet - How it works:
        </h2>
        <div>
          <img
            alt="Secret Spreadsheet Instructions"
            src="assets/img/helper-pic.svg"
          />
          <p>
            Cells of this spreadsheet have been 
            <strong>
              intentionally hidden.
            </strong>
          </p>
          <p>
            Input the hidden data correctly 
            <strong>
              (capitalization matters!)
            </strong>
             to restore the spreadsheet.
          </p>
        </div>
        <button
          onClick={[Function]}
          type="button"
          value="Got it!"
        >
          Got it!
        </button>
      </div>
    </div>
    <div
      className="popup question"
    >
      <div>
        <p
          className="mainQuestion"
        >
          Test Question
        </p>
        <p>
          Test question body
        </p>
        <button
          onClick={[Function]}
          type="button"
          value="Next"
        >
          Next
        </button>
      </div>
    </div>
    <p
      className="instructions"
    >
      Input the 
      <span>
        missing data
      </span>
       to complete the spreadsheet:
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="table-surround"
      >
        <div>
          <table>
            <thead />
            <tbody>
              <tr
                id="row0"
              >
                <td
                  className="centerAlign"
                  id="cell0"
                >
                  <span>
                    Show 1
                  </span>
                </td>
                <td
                  className="unanswered centerAlign"
                  id="cell1"
                >
                  <input
                    className="unanswered"
                    id="1-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
              </tr>
              <tr
                id="row1"
              >
                <td
                  className="unanswered centerAlign"
                  id="cell2"
                >
                  <input
                    className="unanswered"
                    id="2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="?"
                    type="text"
                    value=""
                  />
                </td>
                <td
                  className="centerAlign"
                  id="cell3"
                >
                  <span>
                    Show 2
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <p>
        You've filled out 
        0
         of 
        0
         missing cells
      </p>
      <p
        className="link"
        onClick={[Function]}
      >
        View Question
      </p>
      <input
        className="filled"
        type="submit"
        value="Submit"
      />
    </form>
  </main>
</div>
`;
